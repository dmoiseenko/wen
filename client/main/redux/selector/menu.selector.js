/* eslint-disable import/prefer-default-export */

export const isMenuOpenSelector = state => state.menu.isOpen;

/* eslint-enable */
