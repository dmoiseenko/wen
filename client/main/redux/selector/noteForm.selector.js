import { formValueSelector } from 'redux-form';

import { noteForm, textField } from '../../core/note.form.constant';


export const noteFromSelector = formValueSelector(noteForm);

export const noteTextSelector = state => noteFromSelector(state, textField);
