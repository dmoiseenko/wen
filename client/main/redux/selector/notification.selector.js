export const notifications = state => state.notifications; // eslint-disable-line
