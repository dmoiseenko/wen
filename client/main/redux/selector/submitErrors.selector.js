const validateInput = (state, formName, fieldName) => {
  if (!formName) {
    throw new Error('formName should be defined');
  }

  if (!state) {
    throw new Error('state should be defined');
  }

  if (!fieldName) {
    throw new Error('fieldName should be defined');
  }
};

const checkReduxFormSetup = (state) => {
  if (!state.form) {
    throw new Error('redux-form is not initialized');
  }
};

export default (state, formName, fieldName) => {
  validateInput(state, formName, fieldName);

  checkReduxFormSetup(state);

  if (!state.form[formName]) {
    return '';
  }

  if (!state.form[formName].submitErrors) {
    return '';
  }

  return state.form[formName].submitErrors[fieldName];
};
