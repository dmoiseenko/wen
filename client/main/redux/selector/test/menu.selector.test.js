import { isMenuOpenSelector } from '../menu.selector';


test('isMenuOpenSelector should get isOpen from state', () => {
  const state = {
    menu: {
      isOpen: 'is open'
    }
  };

  expect(isMenuOpenSelector(state)).toMatchSnapshot();
});
