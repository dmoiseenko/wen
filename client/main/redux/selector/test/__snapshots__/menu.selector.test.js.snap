// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isMenuOpenSelector should get isOpen from state 1`] = `"is open"`;
